File: ZeroBit.java
import java.util.*;
public class ZeroBit {
    static  double countStrings(double n) {
        double endWithZero[] = new double [(int)n];
        double endWithOne[] = new double [(int)n];
        endWithZero[0] = endWithOne[0] = 1;
        int k = (int)n;
        for (int i = 1; i < k; i++) {
            endWithZero[i] = endWithZero[i - 1] + endWithOne[i - 1];
            endWithOne[i] = endWithZero[i - 1];
        }
        return endWithZero[k - 1] + endWithOne[k - 1];
    }
    /* Driver program to test above function */
    public static void main (String args[]) {
        Scanner scan = new Scanner(System.in);
        int t = scan.nextInt();

        if (t < 0 || t > Math.pow(10, 3))
            System.exit(0);
        double mod = Math.pow(10, 9) + 7;
        double[] result = new double [t];

        for (int i = 0; i < t; i++) {
            double n = scan.nextDouble();

            if (n < 0 || n > Math.pow(10, 15))
                System.exit(0);

            result[i] = countStrings(n) % mod;

        }
        for (int i = 0; i < t; i++)
            System.out.println((int)result[i]);
    }
}

File: ZeroBit.py
import sys

class Solution:
    def withoutConsectiveZeroes(self, n):

        endWithZero = [0] * (n + 1)
        endWithOne = [0] * (n + 1)

        endWithZero[1] = 1
        endWithOne[1] = 1

        for i in range(2, n + 1):
            endWithZero[i] = endWithZero[i-1] + endWithOne[i-1]
            endWithOne[i] = endWithZero[i-1]

        return endWithZero[n] + endWithOne[n]


t = int(input())
if (t < 1 or t >= 10**3):
    sys.exit("Number of Test Cases should be less 1000")


for number in range(t):
    mod = 10**9 + 7
    result = []

    n = int(input())
    if (n < 1 or n >= 10**15):
        sys.exit("Number of binary digits should be less than 10000")
    
    result = Solution().withoutConsectiveZeroes(n) % mod
    print(result)
